common/djangoapps/student/tests/test_long_username_email.py

```diff
@@ -37,13 +37,13 @@ def test_long_username(self):
 
         obj = json.loads(response.content.decode('utf-8'))
         self.assertEqual(
-            obj['value'],
+            obj['username'][0]['user_message'],
```

Comment:
**Note to reviewers:** These test changes are needed since the old registration API returned the validation error in a `value` field instead of the new format of field_name (e.g., `username`) and `user_message`.